src/OptionSettlement.sol

```diff
@@ -264,9 +264,54 @@ contract OptionSettlementEngine is ERC1155 {
         // TODO(Emit events for indexing and frontend)
     }
 
-    function redeem(uint256 claimId) external view {
+    function redeem(uint256 claimId) external {
         require(tokenType[claimId] == Type.Claim, "Token is not an claim");
-        require(!claim[claimId].claimed, "Already Claimed");
+
+        uint256 balance = this.balanceOf(msg.sender, claimId);
+        require(balance == 1, "no claim token");
+
+        Claim storage claimRecord = claim[claimId];
+
+        require(!claimRecord.claimed, "Already Claimed");
+
+        uint256 optionId = claimRecord.option;
+        Option storage optionRecord = option[optionId];
+
+        require(
+            optionRecord.expiryTimestamp <= block.timestamp,
+            "Not expired yet"
+        );
+
+        uint256 rx_amount = optionRecord.exerciseAmount *
```

Comment:
cool. then I'll change both `rx_amount` and `tx_amount` in `write()` and `exercise()` accordingly?

Reply:
Nope, because those actually denote transmitting and receiving tokens, and are correctly named for what they are doing. In this case we are transmitting tokens only, and there are 2 of them.